source/compute_intersections.cc

```diff
@@ -0,0 +1,387 @@
+               dealii::Point<spacedim>(CGAL::to_double(r->vertex(3).x()),
+                                       CGAL::to_double(r->vertex(3).y())),
+               dealii::Point<spacedim>(CGAL::to_double(r->vertex(2).x()),
+                                       CGAL::to_double(r->vertex(2).y()))}};
+
+            return compute_linear_transformation<dim0, spacedim, 4>(
+              dealii::QGauss<dim0>(degree), vertices_array); // 4 points
+          }
+      }
+    else if (n_vertices_cell0 == 4 && n_vertices_cell1 == 3)
+      { // rectangle-triangle
+        dealii::ExcNotImplemented(
+          "Rectangle-Triangle intersection not yet implemented");
+      }
+
+    else if (n_vertices_cell0 == 2 && n_vertices_cell1 == 4)
+      { // segment-rectangle
+        const auto inters =
+          ::internal::compute_intersection_of_cells<2, 4>(vertices_cell0,
+                                                          vertices_cell1);
+
+        if (inters)
+          {
+            if (const auto *s = boost::get<CGAL_Segment>(&*inters))
+              {
+                std::array<dealii::Point<spacedim>, 2> vertices_array{
+                  {dealii::Point<spacedim>(CGAL::to_double(s->vertex(0).x()),
+                                           CGAL::to_double(s->vertex(0).y())),
+                   dealii::Point<spacedim>(CGAL::to_double(s->vertex(1).x()),
+                                           CGAL::to_double(s->vertex(1).y()))}};
+
+                return (s->is_degenerate()) ?
+                         dealii::Quadrature<spacedim>() :
+                         compute_linear_transformation<dim0, spacedim, 2>(
+                           dealii::QGauss<dim0>(degree),
+                           vertices_array); // 2 points
+              }
+            else
+              {
+                return dealii::Quadrature<spacedim>(); // got a simple Point,
+                                                       // return an empty
+                                                       // Quadrature
+              }
+          }
+      }
+
+    return dealii::Quadrature<spacedim>();
+  }
+
+
+  /**
+   * @brief Given two triangulations cached inside `GridTools::Cache` objects, compute all intersections between the two
+   * and return a vector where each entry is a tuple containing iterators to the
+   * respective cells and a `Quadrature<spacedim>` formula to integrate over the
+   * intersection.
+   *
+   * @tparam dim0 Intrinsic dimension of the immersed grid
+   * @tparam dim1 Intrinsic dimension of the ambient grid
+   * @tparam spacedim
+   * @param space_cache
+   * @param immersed_cache
+   * @param degree Degree of the desired quadrature formula
+   * @return std::vector<std::tuple<
+   * typename dealii::Triangulation<dim0, spacedim>::active_cell_iterator,
+   * typename dealii::Triangulation<dim1, spacedim>::active_cell_iterator,
+   * dealii::Quadrature<spacedim>>>
+   */
```

Comment:
Same here. The documentation goes together with the declaration, not in the cc file, but in the h file.